packages/polymorphic/README.md

```diff
@@ -0,0 +1,148 @@
+
+```tsx
+interface MyProps {
+  someProp: string;
+}
+const MyComponent = Polymorphic<MyProps>(({ as, ...rest }) => {
+  const { Component, ref } = usePolymorphic(as);
+  return <Component ref={ref} {...rest} />;
+});
+```
+
+### With Refs
+
+Accepting a forwarded ref to a Polymorphic component is as simple as passing in a ref into the render function (same as `React.forwardRef`).
+
+```tsx
+interface MyProps {
+  someProp: string;
+}
+const MyComponent = Polymorphic<MyProps>(({ as, ...rest }, forwardedRef) => {
+  const { Component } = usePolymorphic(as);
+  return <Component ref={forwardedRef} {...rest} />;
+});
+```
+
+### Inferred `as` prop
+
+Components extended using the `Polymorphic` factory function can be made to infer the `as` prop value based on the `href` passed in.
+Ensure the custom props are wrapped in `InferredPolymorphicProps`, and use the `useInferredPolymorphic` hook.
+Make sure to pass both `as` and a `rest` object (that may contain `href`) into the hook.
+
+```tsx
+export const MyInferredComponent = Polymorphic<
+  InferredPolymorphicProps<MyProps>
+>(({ as, ...rest }) => {
+  const { Component, ref } = useInferredPolymorphic(as, rest);
+  return (
+    <Component ref={ref} {...rest}>
+      {title}
+    </Component>
+  );
+});
+
+//
+
+<MyInferredComponent href="mongodb.design" />; // renders as <a>
+```
+
+## With Emotion `styled` API
+
+`Polymorphic` also supports usage with Emotions `styled` API. To get TypeScript to accept the Polymorphic props you'll need to explicitly type your styled component as `PolymorphicComponentType`.
+
+```tsx
+const StyledPolymorph = styled(Polymorph)`
+  color: hotpink;
+` as PolymorphicComponentType;
+
+// or
+
+const MyStyledComponent = styled(MyComponent)`
+  color: hotpink;
+` as PolymorphicComponentType;
+```
+
+This also works with InferredPolymorphic components
+
+```tsx
+const StyledInferred = styled(MyInferredComponent)`
+  color: hotpink;
+` as InferredPolymorphicComponentType;
+```
+
+Note: TSDocs will not compile for styled polymorphs. This can be remedied by creating a wrapper around the styled function that explicitly returns a PolymorphicComponentType
+
+## Supported (but not recommended) usage
+
+While it is possible to use the `Polymorph` component to extend polymorphic behavior, it can be much more verbose and error prone than using the `Polymorphic` factory function. For completeness, an example of how to do this is provided below:
+
+```tsx
+type MyProps<T extends PolymorphicAs> = PolymorphicPropsWithRef<
+  T,
+  {
+    someProp: string;
+  }
+>;
+
+export const MyComponent = <T extends PolymorphicAs = 'div'>(
+  { as, title, ...rest }: MyProps<T>,
+  forwardedRef: PolymorphicRef<T>,
+) => {
+  return (
+    <Polymorph as={as as PolymorphicAs} ref={forwardedRef} {...rest}>
+      {title}
+    </Polymorph>
+  );
+};
+```
+
+# Prior art
```

Comment:
I'm wondering if you meant to say `artifacts` here, but maybe `References` is clearer?

Reply:
Prior art is a little unclear

Reply:
[Prior art](https://en.wikipedia.org/wiki/Prior_art) is a fairly common industry term afaik

Reply:
it is indeed, just dropped something in our team channel 👀